app/helpers/application_helper/button/ems_network.rb

```diff
@@ -1,5 +1,9 @@
 class ApplicationHelper::Button::EmsNetwork < ApplicationHelper::Button::Basic
   def visible?
-    ::Settings.product.nuage
+    # Nuage is only provider that supports adding NetworkProvider manually
+    # therefore we hide drop-down option completely unless Nuage is enabled.
+    return ::Settings.product.nuage unless @record # add new
+
+    @record.supports?(:ems_network_new) # edit
```

Comment:
@miha-plesko Quick question.
Does this work from the list view?
I think, it won't - because `@record` does not exist in the list view.

I need to do something similar in one other PR and was looking into options of disabling edit in the list view for a particular type of record. 
(sadly, there is no easy way to do this)

Reply:
@AparnaKarve This doesn't work indeed on the list view. However, I remember we're using something similar on other location like this: 

https://github.com/manageiq/manageiq-ui-classic/blob/master/app/controllers/application_controller/ci_processing.rb#L322-L335

What we do there is we loop all selected items and check for each if it supports specific feature. If there is at least one item found that doesn't support the feature, we refuse performing action and fire up popup saying "Operation does not apply to at least one of the selected items" instead.

Something similar should probably also be implemented here. However, I must apologize, but I'm fully loaded and can't work on this ATM - can it wait a week or two?

Reply:
@miha-plesko That may work - although there would be a slight compromise in the UX, I think.
Meaning, you would have to first go through the `Edit` button click, and then it would be determined if the Edit operation is supported or not -- as opposed to the UI intelligently determining beforehand if the Edit button needs to be visible in the first place.

I was thinking an other easier option would be to simply disable/hide Edit in the list view, because all of the Network Providers, (with the exception of Nuage) do not support Edit
...and, display the Edit button for a Nuage Network Provider only in the summary screen.

Reply:
I agree, Edit button would best not appear on list view at all.